lib_test/helper_server.ml

```diff
@@ -98,17 +98,20 @@ module ALPN = struct
     | _ :: xs ->
       first_match l1 xs
 
-  let https_server port =
+  let https_server ?(check_client_cert=false) port =
     let open Lwt.Infix in
     let listen_address = Unix.(ADDR_INET (inet_addr_loopback, port)) in
+    let ca = "./certificates/ca.pem" in
     let cert = "./certificates/server.pem" in
     let priv_key = "./certificates/server.key" in
     Lwt_io.establish_server_with_client_socket
       listen_address
       (fun client_addr fd ->
         let server_ctx = Ssl.create_context Ssl.TLSv1_3 Ssl.Server_context in
         Ssl.disable_protocols server_ctx [ Ssl.SSLv23; Ssl.TLSv1_1 ];
+        Ssl.load_verify_locations server_ctx ca "";
```

Comment:
@Firgeis why did you add this here? Trying to run the tests locally, it seems like openssl is not happy with it.

Reply:
That loads the ca certificate so it can verify the client cert properly. What error are you getting?

Reply:
It was my bad. `dune runtest` works but `dune exec lib_test/test_client.exe` couldn't find the certificates. 

I'm exploring how to make both work with dune-sites

Reply:
Probably related to the relative paths

Reply:
I fixed the paths and now I'm getting an infinite loop running the client certs test. Similar to what I see in #94 (CI just hangs)

Reply:
Retrieving the client certs here also uses relative paths
https://github.com/Firgeis/piaf/blob/fd5e07ceea0ef9d072abc1b39d39229c1300b85f/lib_test/test_client.ml#L225

Reply:
That's not the issue. The problem is that the server never returns if it can't verify the peer. So the try / catch around the client request is useless because the server never really returns a response.

Reply:
Strange, because when I run the following test: https://github.com/anmonteiro/piaf/blob/6e1b433a09caf5a6dc0ae21c0ab2b1b0d5e1fe69/lib_test/test_client.ml#L278 (which is to cover that case when the client doesn't send the client cert and the server verifies it) is passing, at least when running dune test

Reply:
My suggestion was a try block around the reading of the certs not the client request, sorry if I was unclear